kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -628,11 +632,13 @@ protected Optional<LogSegment> getLogSegmentForVersion(
         final long newVersion = deltaVersionsAfterCheckpoint.isEmpty() ?
             newCheckpointOpt.get().version : deltaVersionsAfterCheckpoint.getLast();
 
-        // In the case where `deltasAfterCheckpoint` is empty, `deltas` should still not be empty,
-        // they may just be before the checkpoint version unless we have a bug in log cleanup.
-        if (deltas.isEmpty()) {
-            throw new IllegalStateException(
-                format("Could not find any delta files for version %s", newVersion)
+        // There should be a delta file present for the newVersion that we are loading
+        // (Even if `deltasAfterCheckpoint` is empty, `deltas` should not be)
+        if (deltas.isEmpty() ||
+            FileNames.deltaVersion(deltas.get(deltas.size() - 1).getPath()) < newVersion) {
+            throw new InvalidTableException(
+                tablePath.toString(),
+                String.format("Missing delta file for version %s", newVersion)
```

Comment:
Missing delta log file(s) needs for state construction?

Reply:
Not sure if it matters but this error you actually can reconstruct the state... (since there is a checkpoint file) but we throw an error anyways (in most cases, see the test suite... it's not all cases) because there should be a delta file for the checkpoint version present